core/pkg/resultshandling/printer/v2/junit.go

```diff
@@ -118,96 +119,119 @@ func (junitPrinter *JunitPrinter) ActionPrint(opaSessionObj *cautils.OPASessionO
-
-			testCase.Failure = &testCaseFailure
-		} else if control.GetStatus().IsSkipped() {
-			testCase.SkipMessage = &JUnitSkipMessage{
-				Message: "", // TODO - fill after statusInfo is supported
+			if fixPaths := fixPathsToString(&c); len(fixPaths) > 0 {
+				msg += fmt.Sprintf("Available fix: \n - %s\n", strings.Join(fixPaths, "\n - "))
 			}
+			msg += "\n"
 
+			severityCounter[apis.ControlSeverityToInt(control.GetScoreFactor())] += 1
 		}
+	}
+	return msg
+}
+
+// Every testCase includes a file (even if the file contains several resources)
+func testsCases(results *cautils.OPASessionObj, resourcesResult []resourcesresults.Result) []JUnitTestCase {
+	var testCases []JUnitTestCase
+	testCase := JUnitTestCase{}
+	testCaseFailure := JUnitFailure{}
+	testCaseFailure.Type = "Resources"
+	message := ""
+
+	// severityCounter represents the severities, 0: Unknown, 1: Low, 2: Medium, 3: High, 4: Critical
+	severityCounter := make([]int, 5, 5)
+
+	for i := range resourcesResult {
+		message += "\n===================================================================================================================\n\n"
+		if failedControls := failedControlsToFailureMessage(results, resourcesResult[i].ListControls(), severityCounter); failedControls != "" {
+			message += fmt.Sprintf("Resource: %s\n\n%s", resourceNameToString(results.AllResources[resourcesResult[i].GetResourceID()]), failedControls)
+		}
+	}
+	testCaseFailure.Message += fmt.Sprintf("%s\n%s", getSummaryMessage(severityCounter), message)
+
+	testCase.Failure = &testCaseFailure
+	if testCase.Failure.Message != "" {
 		testCases = append(testCases, testCase)
 	}
+
 	return testCases
 }
 
-func resourceToString(resource workloadinterface.IMetadata) string {
-	sep := "; "
-	s := ""
-	s += fmt.Sprintf("apiVersion: %s", resource.GetApiVersion()) + sep
-	s += fmt.Sprintf("kind: %s", resource.GetKind()) + sep
-	if resource.GetNamespace() != "" {
-		s += fmt.Sprintf("namespace: %s", resource.GetNamespace()) + sep
+func getSummaryMessage(severityCounter []int) string {
+	total := 0
+	severities := ""
+	for i, count := range severityCounter {
+		if apis.SeverityNumberToString(i) == "Unknown" {
+			continue
+		}
+		severities += fmt.Sprintf("%s: %d, ", apis.SeverityNumberToString(i), count)
+		total += count
 	}
-	s += fmt.Sprintf("name: %s", resource.GetName())
-	return s
+	if len(severities) == 0 {
+		return ""
+	}
+	return fmt.Sprintf("Total: %d (%s)", total, severities[:len(severities)-2])
 }
 
-func properties(riskScore float32) []JUnitProperty {
-	return []JUnitProperty{
-		{
-			Name:  "riskScore",
-			Value: fmt.Sprintf("%.2f", riskScore),
-		},
+func resourceNameToString(resource workloadinterface.IMetadata) string {
+	s := ""
+	s += fmt.Sprintf("kind=%s/", resource.GetKind())
+	if resource.GetNamespace() != "" {
+		s += fmt.Sprintf("namespace=%s/", resource.GetNamespace())
 	}
+	s += fmt.Sprintf("name=%s", resource.GetName())
```

Comment:
There is no separator between the different strings

Reply:
There is a separator after the kind and after the namespace.
After the name is not necessary because it is the last.
The output should be `kind/namespace/name`, not `kind/namespace/name/`